test/conformance/parallel/configuration.go

```diff
@@ -0,0 +1,73 @@
+package test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v2"
+)
+
+type yamlTimeDur time.Duration
+
+type MasterOffsetContinuous struct {
+	Enable    bool  `yaml:"enable"`
+	FailFast  bool  `yaml:"failfast"`
+	MinOffset int   `yaml:"minoffset"`
+	MaxOffset int   `yaml:"maxoffset"`
+	Duration  int64 `yaml:"duration"`
+}
+
+type TestCaseA struct {
+	Enable    bool  `yaml:"enable"`
+	FailFast  bool  `yaml:"failfast"`
+	MinOffset int   `yaml:"minoffset"`
+	MaxOffset int   `yaml:"maxoffset"`
+	Duration  int64 `yaml:"duration"`
+}
+
+type TestCaseB struct {
+	Enable    bool  `yaml:"enable"`
+	FailFast  bool  `yaml:"failfast"`
+	MinOffset int   `yaml:"minoffset"`
+	MaxOffset int   `yaml:"maxoffset"`
+	Duration  int64 `yaml:"duration"`
+}
+
+type Configuration struct {
+	MasterOffsetContinuousConfig MasterOffsetContinuous `yaml:"masteroffsetcontinuous"`
+	TestCaseAConfig              TestCaseA              `yaml:"testacontinuous"`
+	TestCaseBConfig              TestCaseB              `yaml:"testbcontinuous"`
+}
+
+var configuration Configuration
+var loaded bool
+
+func loadConfiguration(filename string, conf *Configuration) {
+	yamlFile, err := ioutil.ReadFile(filename)
+	if err != nil {
+		log.Printf("yamlFile.Get err   #%v ", err)
+	}
+	err = yaml.Unmarshal(yamlFile, &conf)
+	if err != nil {
+		log.Fatalf("Unmarshal: %v", err)
+	}
+}
+
+func GetConfiguration() Configuration {
+	if loaded {
+		return configuration
+	}
+	loaded = true
+	path, err := os.Getwd()
+	if err != nil {
+		logrus.Error("can't retrieve path, trying to continue")
+		return configuration
+	}
+	loadConfiguration(fmt.Sprintf("%s/%s", path, "configuration.yml"), &configuration)
```

Comment:
configuration.yml is too generic , can you name it to ptpTestCaseConfig.yaml?
What will be the default config if configuration is not passed?

Reply:
Do we need to consider default config? If the tests are not configured, the test shall not run for the purpose of saving infrastructure costs.

Reply:
mis-configured ? wont the framework  validate. 
Usually default config is required to run basic tests.
These types test  more run in  labs.

Reply:
What do you mean by basic tests? The ones running serially? Those do not require config now.